assets/src/scripts/store/index.js

```diff
@@ -192,166 +222,43 @@ export const Actions = {
-                    .map(ts => ts.variable)
-                    .reduce((vars, v) => {
-                        vars[v] = action.data.variables[v];
-                        return vars;
-                    }, {})
-            ) : [];
-            currentVar = sorted.length > 0 ? sorted[0].oid : null;
-            return {
-                ...state,
-                series: merge({}, state.series, action.data),
-                showSeries: {
-                    ...state.showSeries,
-                    [action.key]: action.show
-                },
-                currentVariableID: state.currentVariableID || currentVar
-            };
-
-        case 'TOGGLE_TIMESERIES':
-            return {
-                ...state,
-                showSeries: {
-                    ...state.showSeries,
-                    [action.key]: action.show
-                }
-            };
-
-        case 'RESET_TIMESERIES':
-            newState = {
-                ...state,
-                showSeries: {
-                    ...state.showSeries,
-                    [action.key]: false
-                },
-                series: {
-                    ...state.series,
-                    request: {
-                        ...(state.series || {}).request
-                    }
-                }
-            };
-            delete newState.series.request[action.key];
-            return newState;
-
-        case 'SHOW_MEDIAN_STATS_LABEL':
-            return {
-                ...state,
-                showMedianStatsLabel : action.show
-            };
-
-        case 'SET_CURSOR_OFFSET':
-            return {
-                ...state,
-                cursorOffset: action.cursorOffset
-            };
-
-        case 'RESIZE_UI':
-            return {
-                ...state,
-                windowWidth: action.windowWidth,
-                width: action.width
-            };
-
-        case 'PARAMETER_CODE_SET':
-            return {
-                ...state,
-                currentVariableID: action.variableID
-            };
-
-        case 'SET_GAGE_HEIGHT_INDEX':
-            return {
-                ...state,
-                gageHeight: state.floodStages[action.gageHeightIndex]
-            };
-
-        case 'SET_GAGE_HEIGHT':
-            return {
-                ...state,
-                gageHeight: getGageHeight(action.gageHeight, state.floodStages)
-            };
-
-        default:
-            return state;
-    }
-};
-
+const appReducer = combineReducers({
+    series,
+    floodData,
+    timeseriesState,
+    floodState,
+    ui
+});
 
 const MIDDLEWARES = [thunk];
 
 
 export const configureStore = function (initialState) {
     initialState = {
         series: {},
-        statisticalMetaData: {
-            beginYear: '',
-            endYear: ''
+        floodData: {
+            stages: [],
+            extent: {}
+        },
+
+        timeseriesState: {
+            showSeries: {
```

Comment:
I think all of these changes are logical and make sense in my head. Though, I seem to recall us making some effort to normalize the state and reduce nesting. That said, I can't think of a more logical to handle the various timeseries options, so this can probably be left as it is.

Reply:
We did flatten 'series' when Dan brought in Normalizr. I saw this effort as more trying to organize  and shape the store to match the components on the page. I think we still need to work on perhaps refactoring the actions and then for sure when need to work on the selectors for the hydrograph. I'm actually right now reworking the flood data and map component but that is way easier than timeseries/hydrograph. I will use a separate PR for that.